### ISSUE TITLE
Random segfault in the tests

### ISSUE THREAD

**Issue by Geod24** [bosagora/agora#127]
Observed on LDC 1.16, Mac OSX:
```
Target /Users/travis/build/bpfkorea/agora/source/scpp/build/libscp.o is up to date, nothing to do
Deserializing composite type ArgWrapper!() which has no serializable fields.
Serializing composite type ArgWrapper!() which has no serializable fields
Running ./build/agora 
0   agora                               0x000000011016b88e _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff5baf2f5a _sigtramp + 26
0   agora                               0x000000011016b88e _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff5baf2f5a _sigtramp + 26
Program exited with code -11
The command "dub test --skip-registry=all --compiler=${DC}" exited with 2.
```

https://travis-ci.com/bpfkorea/agora/jobs/220748509

**Comment by AndrejMitrovic** [comment]
Hmm maybe that's why my code in https://github.com/bpfkorea/agora/pull/119 fails. It's the same platform. https://travis-ci.com/bpfkorea/agora/jobs/221073787

**Comment by AndrejMitrovic** [comment]
BAD Github!

**Comment by denizzzka** [comment]
@Geod24 you provided URL to sucessful green CI run?

**Comment by denizzzka** [comment]
<s>@AndrejMitrovic failed commit 62e9d8fb80341416cb485557597dd5abb9da50a2 is still available? Can't find it at your fork</s>

Found it

**Comment by AndrejMitrovic** [comment]
v0.x.x is failing randomly too.

**Comment by denizzzka** [comment]
<s>Blocked by https://github.com/Geod24/localrest/issues/19</s>

**Comment by denizzzka** [comment]
Reproduced on Linux with:
```
LDC - the LLVM D compiler (1.16.0):
  based on DMD v2.086.1 and LLVM 8.0.0
  built with LDC - the LLVM D compiler (1.16.0)
  Default target: x86_64-unknown-linux-gnu
  Host CPU: westmere
```

**Comment by denizzzka** [comment]
Catched coredump after SIGABRT. Here is about 37 threads and one in this state:

```gdb
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff743f535 in __GI_abort () at abort.c:79
#2  0x0000555555d4bd65 in core.thread.Fiber.allocStack(ulong, ulong) ()
#3  0x0000555555d4bdc3 in _D4core6thread5Fiber6__ctorMFNbDFZvmmZCQBlQBjQBf ()
#4  0x0000555555d34bc1 in std.concurrency.FiberScheduler.spawn(void() delegate) ()
#5  0x0000555555ab64dd in _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl__T7spawnedTCQBpQBmQBk__T8TestNodeTCQCmQCjQCh18TestNetworkManagerZQBpZQCpFSQDz6common6ConfigQhZ__T6handleTSQGlQGh7CommandZQyMFNbQwZv (arg=...)
    at /home/denizzz/Dev/agora/submodules/localrest/source/geod24/LocalRest.d:449
#6  0x0000555555ab6349 in _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl__T7spawnedTCQBpQBmQBk__T8TestNodeTCQCmQCjQCh18TestNetworkManagerZQBpZQCpFSQDz6common6ConfigQhZ9__lambda6MFZ9__lambda5MFNbSQHbQGx7CommandZv (
    cmd=...) at /home/denizzz/Dev/agora/submodules/localrest/source/geod24/LocalRest.d:485
#7  0x0000555555b5cf22 in _D3std11concurrency7Message__T3mapTDFNbS6geod249LocalRest7CommandZvZQBmMFQBmZv (this=<optimized out>, op=...) at /snap/ldc2/108/bin/../include/d/std/concurrency.d:163
#8  0x0000555555b5ccaf in _D3std11concurrency10MessageBox__T3getTS4core4time8DurationTDFNaNbNiNfCQCrQCq15OwnerTerminatedZvTDFNbNfS6geod249LocalRest11TimeCommandZvTDFNaNbNiNfSQBsQBo9FilterAPIZvTDFNbSQCqQCm8ResponseZvTDFNbSQDnQDj7CommandZvZQGwMFQGwMQGfMQEyMQDoMQCoMQBvZ13onStandardMsgMFKSQJyQJx7MessageZb (msg=...) at /snap/ldc2/108/bin/../include/d/std/concurrency.d:1998
#9  0x0000555555b5d523 in _D3std11concurrency10MessageBox__T3getTS4core4time8DurationTDFNaNbNiNfCQCrQCq15OwnerTerminatedZvTDFNbNfS6geod249LocalRest11TimeCommandZvTDFNaNbNiNfSQBsQBo9FilterAPIZvTDFNbSQCqQCm8ResponseZvTDFNbSQDnQDj7CommandZvZQGwMFQGwMQGfMQEyMQDoMQCoMQBvZ4scanMFKSQJoQJn__T4ListTSQKeQKd7MessageZQwZb (list=...) at /snap/ldc2/108/bin/../include/d/std/concurrency.d:2080
#10 0x0000555555b5c9fb in _D3std11concurrency10MessageBox__T3getTS4core4time8DurationTDFNaNbNiNfCQCrQCq15OwnerTerminatedZvTDFNbNfS6geod249LocalRest11TimeCommandZvTDFNaNbNiNfSQBsQBo9FilterAPIZvTDFNbSQCqQCm8ResponseZvTDFNbSQDnQDj7CommandZvZQGwMFQGwMQGfMQEyMQDoMQCoMQBvZb (this=0x7ffff7319f20, _param_0=..., _param_1=..., _param_2=..., _param_3=..., _param_4=..., _param_5=...) at /snap/ldc2/108/bin/../include/d/std/concurrency.d:2157
#11 0x0000555555ab5eff in _D3std11concurrency__T14receiveTimeoutTDFNaNbNiNfCQBwQBv15OwnerTerminatedZvTDFNbNfS6geod249LocalRest11TimeCommandZvTDFNaNbNiNfSQBsQBo9FilterAPIZvTDFNbSQCqQCm8ResponseZvTDFNbSQDnQDj7CommandZvZQGnFS4core4time8DurationQGuQFmQEbQDaQCgZb (duration=..., _param_1=..., _param_2=..., _param_3=..., _param_4=..., _param_5=...) at /snap/ldc2/108/bin/../include/d/std/concurrency.d:872
#12 0x0000555555ab5d3f in _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl__T7spawnedTCQBpQBmQBk__T8TestNodeTCQCmQCjQCh18TestNetworkManagerZQBpZQCpFSQDz6common6ConfigQhZ9__lambda6MFZv ()
    at /home/denizzz/Dev/agora/submodules/localrest/source/geod24/LocalRest.d:467
#13 0x0000555555d34e2e in std.concurrency.FiberScheduler.create(void() delegate).wrap() ()
#14 0x0000555555d4bbf2 in fiber_entryPoint ()
#15 0x0000000000000000 in ?? ()
```

Looks like too many fibers is spawned sometimes and stack will gone.

**Comment by Geod24** [comment]
Wow, interesting. We could try allocating more pages with the scheduler. How did you manage to reliably reproduce ?

**Comment by denizzzka** [comment]
Rephrase: the free space on the stack ends.

> How did you manage to reliably reproduce ?

Not very reliable: 1/20 chances to reproduce at each start.

But I saved binary and core dump (3Gb)

**Comment by denizzzka** [comment]
But what is purpose of creating such a large number of fibers? While spawing there is no reliable way to find out how many of them have already been created; only counting. In other words, fibers are not very designed for such use.

**Comment by Geod24** [comment]
From experience, fibers can scale in the thousands without problem.

**Comment by AndrejMitrovic** [comment]
> From experience, fibers can scale in the thousands without problem.

However we should check how much space they allocate by default. 

Btw, did the `"--END OF FIBER"` thing got merged into druntime?

**Comment by Geod24** [comment]
> However we should check how much space they allocate by default.

4 pages on Linux.

> Btw, did the "--END OF FIBER" thing got merged into druntime?

Yes

**Comment by AndrejMitrovic** [comment]
> 4 pages on Linux.

But it's the mac that's failing right?

**Comment by Geod24** [comment]
4 pages on `Posix`, sorry

**Comment by AndrejMitrovic** [comment]
I got a different one just now, it seems to happened in a collection cycle:

```
0   agora                               0x000000010aa25fbe _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff50caef5a _sigtramp + 26
2   agora                               0x000000010aa33bbb _D2gc4impl12conservativeQw3Gcx__T4markVbi0ZQkMFNbNlPvQcZv + 171
3   agora                               0x000000010aa29893 _D4core6thread15scanAllTypeImplFNbMDFNbEQBmQBk8ScanTypePvQcZvQgZv + 163
4   agora                               0x000000010aa27629 _D4core6thread18callWithStackShellFNbMDFNbPvZvZv + 41
5   agora                               0x000000010aa29995 thread_scanAll + 53
6   agora                               0x000000010aa30fc3 _D2gc4impl12conservativeQw3Gcx11fullcollectMFNbbZm + 339
7   agora                               0x000000010aa32d43 _D2gc4impl12conservativeQw3Gcx10smallAllocMFNbmKmkxC8TypeInfoZPv + 211
8   agora                               0x000000010aa2eed9 _D2gc4impl12conservativeQw14ConservativeGC__T9runLockedS_DQCeQCeQCcQCnQBs12mallocNoSyncMFNbmkKmxC8TypeInfoZPvS_DQEgQEgQEeQEp10mallocTimelS_DQFiQFiQFgQFr10numMallocslTmTkTmTxQCzZQFcMFNbKmKkKmKxQDsZQDl + 89
9   agora                               0x000000010aa31bb3 _DThn16_2gc4impl12conservativeQw14ConservativeGC6qallocMFNbmkxC8TypeInfoZS4core6memory8BlkInfo_ + 83
10  agora                               0x000000010aa376fd gc_qalloc + 29
11  agora                               0x000000010aa259b2 _D4core6memory2GC6qallocFNaNbmkxC8TypeInfoZSQBqQBo8BlkInfo_ + 18
12  agora                               0x000000010a408644 _D3std5array__T8AppenderTAyaZQo13ensureAddableMFmZ9__lambda3MFNaNbNeZS4core6memory8BlkInfo_ + 36
13  agora                               0x000000010a40830b _D3std5array__T8AppenderTAyaZQo13ensureAddableMFNaNbNfmZv + 859
14  agora                               0x000000010a4091d3 _D3std5array__T8AppenderTAyaZQo__T3putTAaZQiMFQhZ10bigDataFunMFNaNbNfmZQBg + 51
15  agora                               0x000000010a408ff8 _D3std5array__T8AppenderTAyaZQo__T3putTAaZQiMFNaNbNfQnZv + 40
16  agora                               0x000000010a3d513e _D3std5array__T8AppenderTAyaZQo__T3putTwZQhMFNaNfwZv + 94
17  agora                               0x000000010a3d4948 _D4vibe4data4json__T12jsonUnescapeTAyaZQtFNaNfKQmZQp + 680
18  agora                               0x000000010a3d43c5 _D4vibe4data4json__T14skipJsonStringTAyaZQvFNaNfKQmPiZQr + 117
19  agora                               0x000000010a3e1659 _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readValueTSQCpQCn13serialization__T6TraitsTS6geod247bitblob__T7BitBlobVmi512ZQpSQFgQFeQCr13DefaultPolicyZQCsTQEuZQEkMFNfZQFg + 137
20  agora                               0x000000010a3e15ab _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS6geod247bitblob__T7BitBlobVmi512ZQpZQClFNfKQFsZQBw + 27
21  agora                               0x000000010a3e1584 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS6geod247bitblob__T7BitBlobVmi512ZQpZQCeFNfKQFlZQBw + 20
22  agora                               0x000000010a3ebd3c _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS6geod247bitblob__T7BitBlobVmi512ZQpZQCmFKQFrZ9__lambda3MFNfZv + 60
23  agora                               0x000000010a3ebcbf _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readArrayTSQCpQCn13serialization__T6TraitsTAS6geod247bitblob__T7BitBlobVmi512ZQpSQFhQFfQCs13DefaultPolicyZQCtZQEhMFNfMDFNfmZvMDFNfZvZv + 319
24  agora                               0x000000010a3ebb5d _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS6geod247bitblob__T7BitBlobVmi512ZQpZQCmFNfKQFtZQBx + 77
25  agora                               0x000000010a3eac89 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTAS6geod247bitblob__T7BitBlobVmi512ZQpZQCfFNfKQFmZQBx + 9
26  agora                               0x000000010a3eab3c _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl5BlockQgZQCdFKQFiZ__T9__lambda2TQEoZQpMFNfQEyZv + 444
27  agora                               0x000000010a3ea96d _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T14readDictionaryTSQCvQCt13serialization__T6TraitsTS5agora9consensusQEn5BlockQgSQFeQFcQCj13DefaultPolicyZQCkZQEeMFNfMDFNfQFeZvZv + 493
28  agora                               0x000000010a3ea5fe _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl5BlockQgZQCdFNfKQFkZQBo + 174
29  agora                               0x000000010a3ea524 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS5agora9consensusQGe5BlockQgZQBwFNfKQFdZQBo + 20
30  agora                               0x000000010a3ea4c2 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTxS5agora9consensusQGm5BlockQgZQCeFNfKQFlZxQBp + 50
31  agora                               0x000000010a3ea414 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTxS5agora9consensusQGf5BlockQgZQBxFNfKQFeZxQBp + 20
32  agora                               0x000000010a3ea35e _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAxS5agora9consensusQGn5BlockQgZQCfFKQFkZ9__lambda3MFNfZv + 78
33  agora                               0x000000010a3ea2cf _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readArrayTSQCpQCn13serialization__T6TraitsTAxS5agora9consensusQEj5BlockQgSQFaQEyQCl13DefaultPolicyZQCmZQEaMFNfMDFNfmZvMDFNfZvZv + 319
34  agora                               0x000000010a3ea16d _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAxS5agora9consensusQGn5BlockQgZQCfFNfKQFmZQBq + 77
35  agora                               0x000000010a3ea119 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTAxS5agora9consensusQGg5BlockQgZQByFNfKQFfZQBq + 9
36  agora                               0x000000010a3ea0fd _D4vibe4data13serialization__T21deserializeWithPolicyTSQCbQBz4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDyQDwQDu13DefaultPolicyTAxS5agora9consensusQFl5BlockQgTQCqZQFeFNfQDaZQBt + 61
37  agora                               0x000000010a3ea0ad _D4vibe4data13serialization__T11deserializeTSQBrQBp4json__T20JsonStringSerializerTAyaVbi0ZQBfTAxS5agora9consensusQEc5BlockQgTQBrZQDvFNfQCbZQBt + 29
38  agora                               0x000000010a38ed3d _D4vibe4data4json__T15deserializeJsonTAxS5agora9consensusQBy5BlockQgTAyaZQCbFNfQkZQBs + 29
39  agora                               0x000000010a395913 _D6geod249LocalRest__T9RemoteAPITC5agora4node3APIQeZQBe13getBlocksFromMFNfmmZAxSQBu9consensus4data5BlockQg + 659
40  agora                               0x000000010a3c0eaf _D5agora7network13NetworkClientQp13getBlocksFromMFmmZ12__dgliteral3MFNaNiNfZAxSQCz9consensus4data5BlockQg + 31
41  agora                               0x000000010a3c0d9c _D5agora7network13NetworkClientQp__T14attemptRequestVE4vibe4core3log8LogLeveli6TAxSQDd9consensus4data5BlockQgZQCwMFLQBkC9ExceptionZQBz + 76
42  agora                               0x000000010a3b973f _D5agora7network13NetworkClientQp13getBlocksFromMFmmZAxSQCc9consensus4data5BlockQg + 79
43  agora                               0x000000010a3b8f5e _D5agora7network14NetworkManagerQq13getBlocksFromMFNbmMDFNfAxSQCi9consensus4data5BlockQgZbZv + 590
44  agora                               0x000000010a3b8ccd _D5agora7network14NetworkManagerQq20retrieveLatestBlocksMFCQCf4node6LedgerQhZ9__lambda2MFZv + 77
45  agora                               0x000000010a90e30e _D3std11concurrency14FiberScheduler6createMFNbDFZvZ4wrapMFZv + 14
46  agora                               0x000000010aa29f84 fiber_entryPoint + 84
```

**Comment by AndrejMitrovic** [comment]
Another similar one:

```
0   agora                               0x0000000108308fbe _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff79b41f5a _sigtramp + 26
2   agora                               0x0000000108315eb3 _D2gc4impl12conservativeQw3Gcx10smallAllocMFNbmKmkxC8TypeInfoZPv + 579
3   agora                               0x0000000108311ed9 _D2gc4impl12conservativeQw14ConservativeGC__T9runLockedS_DQCeQCeQCcQCnQBs12mallocNoSyncMFNbmkKmxC8TypeInfoZPvS_DQEgQEgQEeQEp10mallocTimelS_DQFiQFiQFgQFr10numMallocslTmTkTmTxQCzZQFcMFNbKmKkKmKxQDsZQDl + 89
4   agora                               0x0000000108314bb3 _DThn16_2gc4impl12conservativeQw14ConservativeGC6qallocMFNbmkxC8TypeInfoZS4core6memory8BlkInfo_ + 83
5   agora                               0x000000010831a6fd gc_qalloc + 29
6   agora                               0x00000001083089b2 _D4core6memory2GC6qallocFNaNbmkxC8TypeInfoZSQBqQBo8BlkInfo_ + 18
7   agora                               0x0000000108328b2f _D2rt8lifetime12__arrayAllocFmKS4core6memory8BlkInfo_xC8TypeInfoxQlZQBl + 303
8   agora                               0x000000010832c344 _d_arrayappendcTX + 1684
9   agora                               0x0000000107c7c86d _D5agora4node6LedgerQh11acceptBlockMFNbNfKxSQBq9consensus4data5BlockQgZb + 253
10  agora                               0x0000000107c7d7ac _D5agora4node6LedgerQh9makeBlockMFNfZv + 108
11  agora                               0x0000000107c7cf01 _D5agora4node6LedgerQh17acceptTransactionMFNfSQBs9consensus4data11TransactionQnZb + 225
12  agora                               0x0000000107c7c250 _D5agora4node14GossipProtocolQq18receiveTransactionMFNfSQCc9consensus4data11TransactionQnZv + 192
13  agora                               0x0000000107c6313d _D5agora4node4NodeQf14putTransactionMFNfSQBn9consensus4data11TransactionQnZv + 77
14  agora                               0x0000000107c6e9a5 _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl13handleCommandFSQCzQCv7CommandQCiSQDrQDn9FilterAPIZv + 4933
15  agora                               0x0000000107c74e4f _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl__T7spawnedTCQBpQBmQBk8TestNodeZQBdFSQCn6common6ConfigQhZ__T6handleTSQEzQEv7CommandZQyMFQuZ9__lambda2MFZv + 175
16  agora                               0x00000001081f130e _D3std11concurrency14FiberScheduler6createMFNbDFZvZ4wrapMFZv + 14
17  agora                               0x000000010830cf84 fiber_entryPoint + 84
```

**Comment by Geod24** [comment]
Similar to @AndrejMitrovic 's, but happens in finalization (!)

```
Running ./build/agora 
0   agora                               0x000000010ee157de _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff5b67bf5a _sigtramp + 26
2   agora                               0x000000010ee36c3d rt_finalize2 + 61
3   agora                               0x000000010ee24491 _D2gc4impl12conservativeQw3Gcx5sweepMFNbZm + 1201
4   agora                               0x000000010ee208bb _D2gc4impl12conservativeQw3Gcx11fullcollectMFNbbZm + 555
5   agora                               0x000000010ee22563 _D2gc4impl12conservativeQw3Gcx10smallAllocMFNbmKmkxC8TypeInfoZPv + 211
6   agora                               0x000000010ee1e6f9 _D2gc4impl12conservativeQw14ConservativeGC__T9runLockedS_DQCeQCeQCcQCnQBs12mallocNoSyncMFNbmkKmxC8TypeInfoZPvS_DQEgQEgQEeQEp10mallocTimelS_DQFiQFiQFgQFr10numMallocslTmTkTmTxQCzZQFcMFNbKmKkKmKxQDsZQDl + 89
7   agora                               0x000000010ee213d3 _DThn16_2gc4impl12conservativeQw14ConservativeGC6qallocMFNbmkxC8TypeInfoZS4core6memory8BlkInfo_ + 83
8   agora                               0x000000010ee26f1d gc_qalloc + 29
9   agora                               0x000000010ee151d2 _D4core6memory2GC6qallocFNaNbmkxC8TypeInfoZSQBqQBo8BlkInfo_ + 18
10  agora                               0x000000010e7f7d44 _D3std5array__T8AppenderTAyaZQo13ensureAddableMFmZ9__lambda3MFNaNbNeZS4core6memory8BlkInfo_ + 36
11  agora                               0x000000010e7f7a0b _D3std5array__T8AppenderTAyaZQo13ensureAddableMFNaNbNfmZv + 859
12  agora                               0x000000010e7f88d3 _D3std5array__T8AppenderTAyaZQo__T3putTAaZQiMFQhZ10bigDataFunMFNaNbNfmZQBg + 51
13  agora                               0x000000010e7f86f8 _D3std5array__T8AppenderTAyaZQo__T3putTAaZQiMFNaNbNfQnZv + 40
14  agora                               0x000000010e7c49de _D3std5array__T8AppenderTAyaZQo__T3putTwZQhMFNaNfwZv + 94
15  agora                               0x000000010e7c41e8 _D4vibe4data4json__T12jsonUnescapeTAyaZQtFNaNfKQmZQp + 680
16  agora                               0x000000010e7c3c65 _D4vibe4data4json__T14skipJsonStringTAyaZQvFNaNfKQmPiZQr + 117
17  agora                               0x000000010e7d0ef9 _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readValueTSQCpQCn13serialization__T6TraitsTS6geod247bitblob__T7BitBlobVmi512ZQpSQFgQFeQCr13DefaultPolicyZQCsTQEuZQEkMFNfZQFg + 137
18  agora                               0x000000010e7d0e4b _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS6geod247bitblob__T7BitBlobVmi512ZQpZQClFNfKQFsZQBw + 27
19  agora                               0x000000010e7d0e24 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS6geod247bitblob__T7BitBlobVmi512ZQpZQCeFNfKQFlZQBw + 20
20  agora                               0x000000010e7d28f4 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl11Transaction5InputZQCoFKQFtZ__T9__lambda2TQEzZQpMFNfQFjZv + 180
21  agora                               0x000000010e7d282d _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T14readDictionaryTSQCvQCt13serialization__T6TraitsTS5agora9consensusQEn11Transaction5InputSQFpQFnQCu13DefaultPolicyZQCvZQEpMFNfMDFNfQFpZvZv + 493
22  agora                               0x000000010e7d24be _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl11Transaction5InputZQCoFNfKQFvZQBz + 174
23  agora                               0x000000010e7d2394 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS5agora9consensusQGe11Transaction5InputZQChFNfKQFoZQBz + 20
24  agora                               0x000000010e7d22de _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS5agora9consensusQGm11Transaction5InputZQCpFKQFuZ9__lambda3MFNfZv + 78
25  agora                               0x000000010e7d224f _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readArrayTSQCpQCn13serialization__T6TraitsTAS5agora9consensusQEi11Transaction5InputSQFkQFiQCv13DefaultPolicyZQCwZQEkMFNfMDFNfmZvMDFNfZvZv + 319
26  agora                               0x000000010e7d20ed _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS5agora9consensusQGm11Transaction5InputZQCpFNfKQFwZQCa + 77
27  agora                               0x000000010e7d1e59 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTAS5agora9consensusQGf11Transaction5InputZQCiFNfKQFpZQCa + 9
28  agora                               0x000000010e7d1d8a _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl11TransactionQnZQCkFKQFpZ__T9__lambda2TQEvZQpMFNfQFfZv + 122
29  agora                               0x000000010e7d1cfd _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T14readDictionaryTSQCvQCt13serialization__T6TraitsTS5agora9consensusQEn11TransactionQnSQFlQFjQCq13DefaultPolicyZQCrZQElMFNfMDFNfQFlZvZv + 493
30  agora                               0x000000010e7d1a18 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl11TransactionQnZQCkFNfKQFrZQBv + 120
31  agora                               0x000000010e7d1994 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS5agora9consensusQGe11TransactionQnZQCdFNfKQFkZQBv + 20
32  agora                               0x000000010e7db2bc _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS5agora9consensusQGm11TransactionQnZQClFKQFqZ9__lambda3MFNfZv + 60
33  agora                               0x000000010e7db23f _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readArrayTSQCpQCn13serialization__T6TraitsTAS5agora9consensusQEi11TransactionQnSQFgQFeQCr13DefaultPolicyZQCsZQEgMFNfMDFNfmZvMDFNfZvZv + 319
34  agora                               0x000000010e7db0dd _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAS5agora9consensusQGm11TransactionQnZQClFNfKQFsZQBw + 77
35  agora                               0x000000010e7da4c9 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTAS5agora9consensusQGf11TransactionQnZQCeFNfKQFlZQBw + 9
36  agora                               0x000000010e7da364 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl5BlockQgZQCdFKQFiZ__T9__lambda2TQEoZQpMFNfQEyZv + 324
37  agora                               0x000000010e7da20d _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T14readDictionaryTSQCvQCt13serialization__T6TraitsTS5agora9consensusQEn5BlockQgSQFeQFcQCj13DefaultPolicyZQCkZQEeMFNfMDFNfQFeZvZv + 493
38  agora                               0x000000010e7d9e9e _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTS5agora9consensusQGl5BlockQgZQCdFNfKQFkZQBo + 174
39  agora                               0x000000010e7d9dc4 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTS5agora9consensusQGe5BlockQgZQBwFNfKQFdZQBo + 20
40  agora                               0x000000010e7d9d62 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTxS5agora9consensusQGm5BlockQgZQCeFNfKQFlZxQBp + 50
41  agora                               0x000000010e7d9cb4 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTxS5agora9consensusQGf5BlockQgZQBxFNfKQFeZxQBp + 20
42  agora                               0x000000010e7d9bfe _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAxS5agora9consensusQGn5BlockQgZQCfFKQFkZ9__lambda3MFNfZv + 78
43  agora                               0x000000010e7d9b6f _D4vibe4data4json__T20JsonStringSerializerTAyaVbi0ZQBf__T9readArrayTSQCpQCn13serialization__T6TraitsTAxS5agora9consensusQEj5BlockQgSQFaQEyQCl13DefaultPolicyZQCmZQEaMFNfMDFNfmZvMDFNfZvZv + 319
44  agora                               0x000000010e7d9a0d _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T23deserializeValueDeducedTAxS5agora9consensusQGn5BlockQgZQCfFNfKQFmZQBq + 77
45  agora                               0x000000010e7d99b9 _D4vibe4data13serialization__T20deserializeValueImplTSQCaQBy4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDxQDvQDt13DefaultPolicyZ__T16deserializeValueTAxS5agora9consensusQGg5BlockQgZQByFNfKQFfZQBq + 9
46  agora                               0x000000010e7d999d _D4vibe4data13serialization__T21deserializeWithPolicyTSQCbQBz4json__T20JsonStringSerializerTAyaVbi0ZQBfSQDyQDwQDu13DefaultPolicyTAxS5agora9consensusQFl5BlockQgTQCqZQFeFNfQDaZQBt + 61
47  agora                               0x000000010e7d994d _D4vibe4data13serialization__T11deserializeTSQBrQBp4json__T20JsonStringSerializerTAyaVbi0ZQBfTAxS5agora9consensusQEc5BlockQgTQBrZQDvFNfQCbZQBt + 29
48  agora                               0x000000010e780f9d _D4vibe4data4json__T15deserializeJsonTAxS5agora9consensusQBy5BlockQgTAyaZQCbFNfQkZQBs + 29
49  agora                               0x000000010e776a23 _D6geod249LocalRest__T9RemoteAPITC5agora4test4Base7TestAPIZQBl13getBlocksFromMFNfmmZAxSQCb9consensus4data5BlockQg + 659
50  agora                               0x000000010e77c7ec _D5agora4test4Base12getAllBlocksFCQBgQBdQBb7TestAPIZAxSQCb9consensus4data5BlockQg + 76
51  agora                               0x000000010e773e14 _D5agora4test4Base__T17containSameBlocksTC6geod249LocalRest__T9RemoteAPITCQCuQCrQCp7TestAPIZQBeZQCxFAQCimZb + 68
52  agora                               0x000000010e771b57 _D5agora4test14NetworkManager17__unittest_L54_C1FZv + 775
53  agora                               0x000000010e77445b _D5agora4test14NetworkManager10__unittestZ + 11
54  agora                               0x000000010ee1585b _D4core7runtime18runModuleUnitTestsUZ14__foreachbody2MFPS6object10ModuleInfoZi + 27
55  agora                               0x000000010ee3b5d2 _D2rt5minfo17moduleinfos_applyFMDFyPS6object10ModuleInfoZiZ14__foreachbody2MFKSQCz19sections_elf_shared3DSOZi + 50
56  agora                               0x000000010ee3c0ea _D2rt19sections_elf_shared3DSO7opApplyFMDFKSQBqQBqQyZiZi + 58
57  agora                               0x000000010ee3b57c _D2rt5minfo17moduleinfos_applyFMDFyPS6object10ModuleInfoZiZi + 28
58  agora                               0x000000010ee28ebf _D6object10ModuleInfo7opApplyFMDFPSQBhQBdZiZi + 31
59  agora                               0x000000010ee15610 runModuleUnitTests + 208
60  agora                               0x000000010ee3327f _D2rt6dmain211_d_run_mainUiPPaPUAAaZiZ6runAllMFZv + 31
61  agora                               0x000000010ee330c3 _d_run_main + 547
62  agora                               0x000000010e78aff5 main + 37
Program exited with code -11
The command "dub test -b unittest-cov --skip-registry=all --compiler=${DC}" exited with 2.
```

**Comment by Geod24** [comment]
```
"Request timed-out"
Error: clean-up of Database incorrectly depends on destructors called by the GC.
Info: :memory:
core.exception.InvalidMemoryOperationError@core/exception.d(702): Invalid memory operation
----------------
"Request timed-out"
"Request timed-out"
"Request timed-out"
0   agora                               0x00000001023e234e _D4core7runtime18runModuleUnitTestsUZ19unittestSegvHandlerUNbiPSQCk3sys5posix6signal9siginfo_tPvZv + 62
1   libsystem_platform.dylib            0x00007fff7f341f5a _sigtramp + 26
2   agora                               0x00000001024060dd rt_finalize2 + 61
3   agora                               0x00000001023f18e1 _D2gc4impl12conservativeQw3Gcx5sweepMFNbZm + 1201
4   agora                               0x00000001023ed939 _D2gc4impl12conservativeQw3Gcx11fullcollectMFNbbZm + 585
5   agora                               0x00000001023ef6a3 _D2gc4impl12conservativeQw3Gcx10smallAllocMFNbmKmkxC8TypeInfoZPv + 211
6   agora                               0x00000001023eb729 _D2gc4impl12conservativeQw14ConservativeGC__T9runLockedS_DQCeQCeQCcQCnQBs12mallocNoSyncMFNbmkKmxC8TypeInfoZPvS_DQEgQEgQEeQEp10mallocTimelS_DQFiQFiQFgQFr10numMallocslTmTkTmTxQCzZQFcMFNbKmKkKmKxQDsZQDl + 89
7   agora                               0x00000001023ee463 _DThn16_2gc4impl12conservativeQw14ConservativeGC6qallocMFNbmkxC8TypeInfoZS4core6memory8BlkInfo_ + 83
8   agora                               0x00000001023f633d gc_qalloc + 29
9   agora                               0x00000001023e1d42 _D4core6memory2GC6qallocFNaNbmkxC8TypeInfoZSQBqQBo8BlkInfo_ + 18
10  agora                               0x0000000102406599 _d_arraysetlengthT + 745
11  agora                               0x0000000101d793b7 _D5agora9consensus4data11TransactionQn11deserializeMFNbNfMDFNbNfmZAhZv + 71
12  agora                               0x0000000101d7bca8 _D5agora6common12Deserializer__T15deserializePartTSQBx9consensus4data11TransactionQnZQCbFNbNfKQBsMDFNbNfmZAhZv + 40
13  agora                               0x0000000101d7bb9a _D5agora9consensus4data5BlockQg11deserializeMFNbNfMDFNbNfmZAhZv + 186
14  agora                               0x0000000101db7e8f _D5agora6common12Deserializer__T11deserializeTSQBt9consensus4data5BlockQgZQBqFNbNfMAhZQBo + 95
15  agora                               0x0000000101e3d965 _D5agora6common12BlockStorage15MemBlockStorage9readBlockMFNbNfKSQCk9consensus4data5BlockQgmZb + 373
16  agora                               0x0000000101e3d7d9 _D5agora6common12BlockStorage15MemBlockStorage13readLastBlockMFNbNfKSQCp9consensus4data5BlockQgZb + 105
17  agora                               0x0000000101e3e05d _DThn16_5agora6common12BlockStorage15MemBlockStorage13readLastBlockMFNbNfKSQCp9consensus4data5BlockQgZb + 13
18  agora                               0x0000000101d50df3 _D5agora4node6LedgerQh14getBlockHeightMFNbNfZm + 99
19  agora                               0x0000000101d70cff _D5agora7network14NetworkManagerQq20retrieveLatestBlocksMFCQCf4node6LedgerQhZ9__lambda2MFZv + 47
20  agora                               0x00000001022c941e _D3std11concurrency14FiberScheduler6createMFNbDFZvZ4wrapMFZv + 14
21  agora                               0x00000001023e66f4 fiber_entryPoint + 84
Program exited with code -11
The command "./ci/run.sh" exited with 2.
```

**Comment by Geod24** [comment]
This SEGV is now triggering on every PR...

**Comment by Geod24** [comment]
This has been fixed
